llvm/lib/CodeGen/StackMaps.cpp

```diff
@@ -741,8 +742,19 @@ void StackMaps::emitCallsiteEntries(MCStreamer &OS) {
     OS.emitInt16(0);
     OS.emitInt16(CSLiveVars.size()); // Num lives.
 
+    size_t LiveIdx = 0;
     for (const auto &LiveVar : CSLiveVars) {
+      // YKFIXME: For now the number of locations per live variable is
+      // expressed with a single byte.
+      size_t NumLocs = LiveVar.size();
+      if (NumLocs > std::numeric_limits<uint8_t>::max()) {
+        std::stringstream Msg;
+        Msg << "stackmap ID " << CSI.ID
+            << ": too many locations in live variable #" << LiveIdx;
+        OS.getContext().reportError(SMLoc(), Msg.str());
```

Comment:
Should we `return` at this point or ... ?

Reply:
Yes, we should. It prevents code from crashing later and spamming the console with a backtrace! Good catch.

Fixed in 0896c8ff40bf